### DISCUSSION TITLE
When building all pages, is there a way to catch pages that are on build error

### DISCUSSION THREAD

**Discussion by RodolpheNivelet** [gatsbyjs/gatsby#35634] [Help] [answered]
Hello,

I manage a website with more than 1000 pages with data coming from DatoCMS, and sometimes page in there are unstable and produce a page build error. Sometimes it is because one value is not present which should be verified directly in Dato you might say, but sometimes it is because WYSIWYG editor (MDX content) contains invalid HTML code (such as not closed tag or so...).

So my question is : Is it possible to catch build error when creating pages and not stop build execution on the first error, so I would be able to report which pages are on error.

Thank you,
Rodolphe Nivelet

**Comment by KyleAMathews** [comment, marked as answer]
If you build with the environment variable `GATSBY_IS_PREVIEW` then it won't crash on these pages — it'll instead write out an HTML file with the error https://github.com/gatsbyjs/gatsby/pull/33184

**Comment by RodolpheNivelet** [comment]
Hello thanks @KyleAMathews for your answer.

I have created a bash script to be able to report page path on error as JSON file :

```
errorFiles=`grep -rl 'public' -e "<h1>Preview build error</h1>"`

errorJsonFilePath="./error-report.json"

if [ -n "$errorFiles" ]; then
    errorJson="["
    while read -r line; do
        filePath=`grep '"path": "' $line`
        filePathCleaned=`sed -e 's/ //g' -e 's/,$//g' <<< $filePath`
        errorJson="$errorJson{$filePathCleaned}," >> $errorJsonFilePath
    done <<< "$errorFiles"
    errorJson=`sed 's/,$//g' <<< $errorJson`
    errorJson="$errorJson]"
    echo $errorJson > $errorJsonFilePath
    exit 1
fi
```
